conn.go

```diff
@@ -41,13 +47,24 @@ func NewConn(conn io.ReadWriteCloser) *Conn {
 	return c
 }
 
-// Dial connects to the given address on the given network using net.Dial
-// and then returns a new Conn for the connection.
+// Dial connects to the given address on the given network using net.DialTimeout
+// with a default timeout of 10s and then returns a new Conn for the connection.
 func Dial(network, addr string) (*Conn, error) {
-	c, err := net.Dial(network, addr)
+	connTimeout := time.Duration(DefaultConnTimeout) * time.Second
+	return DialTimeout(network, addr, connTimeout)
+}
+
+// DialTimeout connects to the given address on the given network using net.DialTimeout
+// with a supplied timeout and then returns a new Conn for the connection.
+func DialTimeout(network, addr string, timeout time.Duration) (*Conn, error) {
+	c, err := net.DialTimeout(network, addr, timeout)
```

Comment:
It might be easier to use a net.Dialer to set the keepalive. Something like

```suggestion
	dialer := &net.Dialer{
		Timeout:   timeout,
		KeepAlive: DefaultKeepAliveTimeout,
	}
	c, err := dialer.Dial(network, addr)
```

And then the rest of this function would be less complex.